zerver/lib/streams.py

```diff
@@ -757,27 +757,57 @@ def can_access_stream_history_by_id(user_profile: UserProfile, stream_id: int) -
     return can_access_stream_history(user_profile, stream)
 
 
-def can_remove_subscribers_from_stream(
-    stream: Stream, user_profile: UserProfile, sub: Subscription | None
+def can_remove_subscribers_from_all_streams(
+    streams: list[Stream], user_profile: UserProfile
 ) -> bool:
-    if not check_basic_stream_access(user_profile, stream, sub, allow_realm_admin=True):
-        return False
-
     # Optimization for the organization administrator code path. We
     # don't explicitly grant realm admins this permission, but admins
     # implicitly have the can_administer_channel_group permission for
-    # all accessible channels.
+    # all channels. For channels the user cannot access, they can
```

Comment:
I'm not sure I like dropping "accessible" here. The "For" clause is good. I think the following would be really clear:

"for all accessible channels. For channels that the administrator cannot access, they can do limited administration including removing subscribers".

Reply:
[Fixed](https://github.com/zulip/zulip/compare/a00f864b7fcce02c82a09c1239a9945cd8eb7b8a..cb6dc12385216271413b6355a4684a00424b9228)